src/commands/build.rs

```diff
@@ -1,51 +1,55 @@
 use std::{ffi::OsString, path::PathBuf};
 
 use anyhow::Result;
-use clap::Parser;
+use clap::{Parser, Args, ArgGroup, CommandFactory, FromArgMatches};
 
 use crate::opts::{APP_CONFIG_FILE_OPT, BUILD_UP_OPT, DEFAULT_MANIFEST_FILE};
 
-use super::up::UpCommand;
+use super::up::{UpCommand, UpArgs};
 
-/// Run the build command for each component.
-#[derive(Parser, Debug)]
-#[clap(about = "Build the Spin application", allow_hyphen_values = true)]
-pub struct BuildCommand {
+
+#[derive(Args, Debug)]
+pub struct BuildArgs {
     /// Path to spin.toml.
-    #[clap(
-            name = APP_CONFIG_FILE_OPT,
-            short = 'f',
-            long = "file",
-        )]
+    #[arg(
+        short = 'f',
+        long = "file",
+        id = APP_CONFIG_FILE_OPT,
+    )]
     pub app: Option<PathBuf>,
+}
 
-    /// Run the application after building.
-    #[clap(name = BUILD_UP_OPT, short = 'u', long = "up")]
+#[derive(Args, Debug)]
+pub struct BuildUpArgs {
+    #[arg(short, long, group = "UpArgs", required=true)]
     pub up: bool,
+    #[command(flatten, next_help_heading="Up")]
+    pub args: UpArgs
+}
```

Comment:
I think this worked and it's far better than I expected.

If we separate UpArgs into its own struct, clap implicitly creates a group with the same name as the struct.

Now all I had to do is create separate group for the up flag with the up args flattened into it and mark up as required and as part of the implicit "UpArgs" group.

<img width="519" alt="Screenshot 2023-02-02 at 20 56 05" src="https://user-images.githubusercontent.com/5606812/216478501-7ea2ddae-c324-4183-aba0-413f1c373da6.png">

Reply:
Oh nice!